flumeserver/indexer/indexer.go

```diff
@@ -1,405 +1,252 @@
-  if compressor == nil {
-    compressor = zlib.NewWriter(compressionBuffer)
-  } else {
-    compressor.Reset(compressionBuffer)
-  }
-  compressor.Write(data)
-  compressor.Close()
-  return compressionBuffer.Bytes()
+	if len(data) == 0 {
+		return data
+	}
+	compressionBuffer.Reset()
+	if compressor == nil {
+		compressor = zlib.NewWriter(compressionBuffer)
+	} else {
+		compressor.Reset(compressionBuffer)
+	}
+	compressor.Write(data)
+	compressor.Close()
+	return compressionBuffer.Bytes()
 }
 
 func getCopy(in []byte) []byte {
-  out := make([]byte, len(in))
-  copy(out, in)
-  return out
+	out := make([]byte, len(in))
+	copy(out, in)
+	return out
 }
 
-func getFuncSig(data []byte) ([]byte) {
-  if len(data) >= 4 {
-    return data[:4]
-  }
-  return data[:len(data)]
+func getFuncSig(data []byte) []byte {
+	if len(data) >= 4 {
+		return data[:4]
+	}
+	return data[:]
 }
 
 func nullZeroAddress(addr common.Address) []byte {
-  if addr == (common.Address{}) {
-    return []byte{}
-  }
-  return addr.Bytes()
+	if addr == (common.Address{}) {
+		return []byte{}
+	}
+	return addr.Bytes()
 }
 
 type bytesable interface {
-  Bytes() []byte
+	Bytes() []byte
 }
 
 // applyParameters applies a set of parameters into a SQL statement in a manner
 // that will be safe for execution. Note that this should only be used in the
 // context of blocks, transactions, and logs - beyond the datatypes used in
 // those datatypes, safety is not guaranteed.
 func applyParameters(query string, params ...interface{}) string {
-  preparedParams := make([]interface{}, len(params))
-  for i, param := range params {
-    switch value := param.(type) {
-    case []byte:
-      if len(value) == 0 {
-        preparedParams[i] = "NULL"
-      } else {
-        preparedParams[i] = fmt.Sprintf("X'%x'", value)
-      }
-    case bytesable:
-      b := trimPrefix(value.Bytes())
-      if len(b) == 0 {
-        preparedParams[i] = "NULL"
-      } else {
-        preparedParams[i] = fmt.Sprintf("X'%x'", b)
-      }
-    case hexutil.Bytes:
-      if len(value) == 0 {
-        preparedParams[i] = "NULL"
-      } else {
-        preparedParams[i] = fmt.Sprintf("X'%x'", []byte(value[:]))
-      }
-    case *hexutil.Big:
-      if value == nil {
-        preparedParams[i] = "NULL"
-      } else {
-        preparedParams[i] = fmt.Sprintf("X'%x'", trimPrefix(value.ToInt().Bytes()))
-      }
-    case hexutil.Uint64:
-      preparedParams[i] = fmt.Sprintf("%v", uint64(value))
-    case types.BlockNonce:
-      preparedParams[i] = fmt.Sprintf("%v", int64(value.Uint64()))
-    default:
-      preparedParams[i] = fmt.Sprintf("%v", value)
-    }
-  }
-  return fmt.Sprintf(query, preparedParams...)
+	preparedParams := make([]interface{}, len(params))
+	for i, param := range params {
+		switch value := param.(type) {
+		case []byte:
+			if len(value) == 0 {
+				preparedParams[i] = "NULL"
+			} else {
+				preparedParams[i] = fmt.Sprintf("X'%x'", value)
+			}
+		case *common.Address:
```

Comment:
Do you remember why we made a special case for *common.Address when it implements the bytesable interface and this seems to be identical to the bytesable case? I seem to recall being involved in that decision, but I don't remember why we did it.